tests/test_socket_server.py

```diff
@@ -36,7 +44,7 @@
 """Standard timeout in seconds."""
 
 
-class SocketServerTestCase(unittest.IsolatedAsyncioTestCase):
+class SocketServerTestCase(BaseMockTestCase):
```

Comment:
What is the instance variable `self.data` for?

I looked through the code and it looks like you only use it as a local variable (and in just one method: `socket_server_test`). I suggest removing it from asyncSetUp and renaming `self.data` to `data` in `socket_server_test`. If you do want to return data read you could it from that method.

Reply:
Done